File: Knapsack/zeroone.py
def knapSack(W, wt, val, n):
    #Base case
    if( W == 0 or n== 0):
        return 0
    # If weight of the nth item is more than Knapsack of capacity 
    # W, then this item cannot be included in the optimal solution 
    if (wt[n-1] > W): 
        return knapSack(W, wt, val, n-1) 
  
    # return the maximum of two cases: 
    # (1) nth item included 
    # (2) not included 
    else: 
        return max(val[n-1] + knapSack(W-wt[n-1], wt, val, n-1), 
                   knapSack(W, wt, val, n-1)) 

# To test above function 
val = [0 , 1 , 2 , 5 , 6] 
wt = [0, 2, 3, 4, 5] 
W = 8
n = len(val) 
print (knapSack(W, wt, val, n)) 
print(n)

File: Knapsack/zeroone.cpp
#include<iostream>
#include<algorithm>
using namespace std;
int max(int i, int j){
    return( i>= j? i: j);
}
int main(){
  int P[5][9] = {0, 1, 2, 5, 6};
  int Wt[5] = {0, 2,3 ,4, 5};
  int m = 8, n = 4;
  int K[5][9];
  for(int i = 0; i<= n ; i++){
      for(int w = 0; w<=m; w++){
          if(i == 0 || w == 0) {
            K[i][w] = 0;
          }
          else if(Wt[i] <= w) {
              K[i][w] = max(P[i] + K[i-1][w- Wt[i]], K[i-1][w]);
              }
          else{
              K[i][w] = K[i-1][w];
          } 
      }
  } 

  cout<<K[n][m]; 
}